.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,46 @@
+# This workflow runs pre-commit against the repository.
+#
+# You can adjust the behavior by modifying this file.
+name: Run pre-commit
+
+on: [pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      matrix:
+        python-version:
+          # TODO When 3.7 is removed add dead to pre-commit-config.yaml
+          - "3.7"  # EOL 2023-06-27
+          - "3.8"  # EOL 2024-10
+          - "3.9"  # EOL 2025-10
+          - "3.10"  # EOL 2026-10
+          - "3.11"  # EOL 2027-10
+        experimental: [false]
+        include:
+          - python-version: 3.x
+            experimental: true
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
Aren't the v2 actions deprecated?

Also any rason why not to use the same configuration that comes from pre-commit (https://github.com/pre-commit/action) that is independent on the Python version? (The pre-commit checks shouldn't depend on the Python version used, no?)

Reply:
I'll give that a shot.  Since I use the environment for the pylint check (less duplication of dependencies), I assumed the default wouldn't work.

As for the version, I expected dependabot to take care of any issues, but I can also bump that.